packages/frontend/src/app/create-store/page.tsx

```diff
@@ -95,16 +95,49 @@ const StoreCreation = () => {
   useEffect(() => {
     if (relayClient && isAuthenticated === IStatus.Complete) {
       (async () => {
+        const publishedTagId = new Uint8Array(32);
+        crypto.getRandomValues(publishedTagId);
         await relayClient.shopManifest({
           name: storeName,
           description,
-          profilePictureUrl: avatar,
+          profilePictureUrl: "https://http.cat/images/200.jpg",
+          publishedTagId,
         });
         console.log("store manifested.");
-        const publishedTagId = await relayClient.createTag("visible");
-        if (publishedTagId) {
-          await relayClient!.updateShopManifest({ publishedTagId });
+        const newPubId = await relayClient.createTag({ name: "visible" });
+        if (newPubId) {
+          await relayClient!.updateShopManifest({ publishedTagId: newPubId });
         }
+        const path = await relayClient!.uploadBlob(avatar as FormData);
+        const metadata = {
+          title: "metadata",
+          type: "object",
+          properties: {
+            name: {
+              type: "string",
+              description: storeName,
+            },
+            description: {
+              type: "string",
+              description,
+            },
+            image: {
+              type: "string",
+              description: path.url,
+            },
+          },
+        };
```

Comment:
The schema/format here needs a little fixing up.

https://eips.ethereum.org/EIPS/eip-721 lists it as "JSON schema" which defines what the fields mean (description) and what type they are.

The _actual_ value would just be a flat object ala

```
{
  "name": "my store",
  "description": "cat nip and other fun things",
  "image": "https://you.get/the/idea.png"
}
```

here is a more elaborate example going from value to schema: https://json-schema.org/learn/getting-started-step-by-step